mysql-test/suite/rpl/t/rpl_semi_sync_cond_var_per_thd.test

```diff
@@ -77,4 +80,116 @@ set @@global.binlog_commit_wait_usec=@save_bgc_usec;
 set @@global.debug_dbug=@save_debug_dbug;
 drop table t1, t2, t3;
 
+
+--echo #
+--echo # MDEV-34122
+--echo # If semi-sync is switched off then on while a transaction is
+--echo # in-between binlogging and waiting for an ACK, ensure that the
+--echo # transaction skips the wait altogether (otherwise it would time-out).
+--echo # Note that prior to MDEV-34122, there was a debug assertion that would
+--echo # trigger if the transaction tried to wait but could not receive an ACK
+--echo # signal.
+
+--echo #
+--echo # MDEV-34122.a: Test wait_point = AFTER_SYNC
+--echo # Here, debug_sync is used to pause the leader thread between reporting
+--echo # the binlogging to semi-sync, and starting the wait for ACK; and during
+--echo # this pause, semi-sync is manually switched off and on.
+
+
+--connection master
+set @old_master_wait_point= @@global.rpl_semi_sync_master_wait_point;
+set @old_master_wait_no_slave= @@global.rpl_semi_sync_master_wait_no_slave;
+set @@global.rpl_semi_sync_master_wait_point= AFTER_SYNC;
+set @@global.rpl_semi_sync_master_wait_no_slave= 1;
+
+--echo # Waiting for semi-sync to turn on..
+let $status_var= rpl_semi_sync_master_status;
+let $status_var_value= ON;
+source include/wait_for_status_var.inc;
+--connection slave
+let $status_var= rpl_semi_sync_slave_status;
+let $status_var_value= ON;
+source include/wait_for_status_var.inc;
+
+--connection master
+create table t (a int) engine=innodb;
+--sync_slave_with_master
+
+--echo # Wait no slave is on, so we don't need to slave on to wait for the
+--echo # ACK, and this way we guarantee the ACK doesn't come, so we can verify
+--echo # the informational message is displayed
+--connection slave
+--source include/stop_slave.inc
+
+--connection server_1
+start transaction;
+insert into t values (0);
+set debug_sync= "commit_after_release_LOCK_log SIGNAL trx_binlogged WAIT_FOR continue_commit";
+--send commit
+
+--connection master
+set debug_sync= "now WAIT_FOR trx_binlogged";
+
+--echo # Switching semi-sync off/on
+set @@global.rpl_semi_sync_master_enabled= 0;
+set @@global.rpl_semi_sync_master_enabled= 1;
+let $status_var= rpl_semi_sync_master_status;
+let $status_var_value= ON;
+source include/wait_for_status_var.inc;
+
+--echo # Resuming transaction to await ACK
+set debug_sync= "now SIGNAL continue_commit";
+
+--connection server_1
+--reap
+
+--let $assert_text= Ensure error log shows that transaction is skipping its semi-sync wait
+--let $assert_select=Skipping semi-sync wait for transaction
+--let $assert_file= $MYSQLTEST_VARDIR/log/mysqld.1.err
+--let $assert_count= 1
+--let $assert_only_after=CURRENT_TEST
+--source include/assert_grep.inc
```

Comment:
This part of the test is not necessary since the information message should be removed.
If you want, and to not waste the great effort on the test case, feel free to use DBUG injection or similar to get a message through in the test case to verify this anyway, if you find it meaningful, up to you.